File: java/Film.java
// deklarasi class Film
public class Film {
    // private atribut
    private int id_film;
    private String judul;
    private String genre;
    private int durasi;  // dalam menit
    private int harga;   // harga tiket dalam rupiah

    // Constructor
    public Film(int id_film_baru, String judul_baru, String genre_baru, int durasi_baru, int harga_baru) {
        this.id_film = id_film_baru; // inisialisasi
        this.judul = judul_baru;     // inisialisasi
        this.genre = genre_baru;     // inisialisasi
        this.durasi = durasi_baru;   // inisialisasi
        this.harga = harga_baru;     // inisialisasi
    }

    // Getter (untuk mendapatkan nilai atribut)
    public int getId() {
        return id_film; // mengembalikan nilai atribut
    }

    public String getJudul() {
        return judul; // mengembalikan nilai atribut
    }

    public String getGenre() {
        return genre; // mengembalikan nilai atribut
    }

    public int getDurasi() {
        return durasi; // mengembalikan nilai atribut
    }

    public int getHarga() {
        return harga; // mengembalikan nilai atribut
    }

    // Setter (untuk mengubah nilai atribut)
    public void setId(int id_film) {
        this.id_film = id_film; // menginisialisasi atribut dengan value baru
    }

    public void setJudul(String judul) {
        this.judul = judul; // menginisialisasi atribut dengan value baru
    }

    public void setGenre(String genre) {
        this.genre = genre; // menginisialisasi atribut dengan value baru
    }

    public void setDurasi(int durasi) {
        if (durasi >= 0) { // durasi harus lebih dari atau sama dengan 0
            this.durasi = durasi; // menginisialisasi atribut dengan value baru
        } else {
            // pesan jika durasi negatif
            System.out.println("Durasi tidak boleh negatif.");
        }
    }

    public void setHarga(int harga) {
        if (harga > 0) { // harga harus lebih dari 0
            this.harga = harga; // menginisialisasi atribut dengan value baru
        } else {
            // pesan jika harga tidak valid
            System.out.println("Harga harus lebih dari 0.");
        }
    }

    // prosedur untuk menampilkan data
    public void tampilkanData() {
        // print data
        System.out.println("ID Film : " + getId());
        System.out.println("Judul : " + getJudul());
        System.out.println("Genre : " + getGenre());
        System.out.println("Durasi : " + getDurasi() + " menit");
        System.out.println("Harga Tiket : Rp. " + getHarga());
    }
}

File: python/Film.py
# deklarasi class Film
class Film:
    # contructor
    def __init__(self, id_film:int, judul:str, genre:str, durasi:int, harga:int):
        self.__id_film:int = id_film
        self.__judul:str = judul
        self.__genre:str = genre
        self.__durasi:int = durasi  # dalam menit
        self.__harga:int = harga    # harga tiket dalam rupiah

    # getter (mengambil data)
    def getId(self) -> int:
        return self.__id_film # mengembalikan nilai

    def getJudul(self) -> str:
        return self.__judul # mengembalikan nilai

    def getGenre(self) -> str:
        return self.__genre # mengembalikan nilai

    def getDurasi(self) -> int:
        return self.__durasi # mengembalikan nilai

    def getHarga(self) -> int:
        return self.__harga # mengembalikan nilai

    # setter (untuk merubah value)
    def setId(self, id_film: int):
        self.__id_film = id_film # mengubah value id_film

    def setJudul(self, judul:str):
        self.__judul = judul # mengubah value atribut dengan value baru

    def setGenre(self, genre:str):
        self.__genre = genre # mengubah value atribut dengan value baru

    def setDurasi(self, durasi:int):
        if durasi >= 0: # durasi harus lebih dari 0 dan tidak boleh negatif
            self.__durasi = durasi # mengubah value atribut dengan value baru
        else:
            print("Durasi tidak boleh negatif.") # pesan jika angka negatif

    def setHarga(self, harga:int):
        if harga > 0: # harga harus lebih dari 0 dan tidak boleh negatif
            self.__harga = harga # mengubah value atribut dengan value baru
        else:
            print("Harga harus lebih dari 0.") # pesan jika angka negatif

    # prosedur untuk menampilkan data
    def tampilkanData(self):
        # print data
        print("ID Film :", self.getId())
        print("Judul :", self.getJudul())
        print("Genre :", self.getGenre())
        print("Durasi :", self.getDurasi(), "menit")
        print("Harga Tiket : Rp.", self.getHarga(),)